CI4/tests/system/Helpers/TextHelperTest.php

```diff
@@ -0,0 +1,294 @@
+                'this is a phrase'          => '<mark>this is</mark> a phrase',
+                'this is another'           => '<mark>this is</mark> another',
+                'Gimme a test, Sally'       => 'Gimme a test, Sally',
+                'Or tell me what this is'   => 'Or tell me what <mark>this is</mark>',
+                ''                          => ''
+        ];
+
+        foreach ($strs as $str => $expect)
+        {
+            $this->assertEquals($expect, highlight_phrase($str, 'this is'));
+        }
+
+        $this->assertEquals('<strong>this is</strong> a strong test', highlight_phrase('this is a strong test', 'this is', '<strong>', '</strong>'));
+    }
+
+    // ------------------------------------------------------------------------
+
+    public function test_ellipsize()
+    {
+        $strs = [
+                '0'		=> [
+                        'this is my string'				=> '&hellip; my string',
+                        "here's another one"			=> '&hellip;nother one',
+                        'this one is just a bit longer'	=> '&hellip;bit longer',
+                        'short'							=> 'short'
+                ],
+                '.5'	=> [
+                        'this is my string'				=> 'this &hellip;tring',
+                        "here's another one"			=> "here'&hellip;r one",
+                        'this one is just a bit longer'	=> 'this &hellip;onger',
+                        'short'							=> 'short'
+                ],
+                '1'	=> [
+                        'this is my string'				=> 'this is my&hellip;',
+                        "here's another one"			=> "here's ano&hellip;",
+                        'this one is just a bit longer'	=> 'this one i&hellip;',
+                        'short'							=> 'short'
+                ],
+        ];
+
+        foreach ($strs as $pos => $s)
+        {
+            foreach ($s as $str => $expect)
+            {
+                $this->assertEquals($expect, ellipsize($str, 10, $pos));
+            }
+        }
+    }
+
+    // ------------------------------------------------------------------------
+
+    public function test_word_wrap()
+    {
+        $string = 'Here is a simple string of text that will help us demonstrate this function.';
+        $this->assertEquals(substr_count(word_wrap($string, 25), "\n"), 4);
+    }
+
+    // ------------------------------------------------------------------------
+
+    public function test_default_word_wrap_charlim()
+    {
+        $string = "Here is a longer string of text that will help us demonstrate the default charlim of this function.";
+        $this->assertEquals(strpos(word_wrap($string), "\n"), 73);
+    }
+    
+    // -----------------------------------------------------------------------
+    
+    public function test_excerpt()
+    {
+        $string = $this->_long_string;
+        $phrase = 'write';
+        $this->assertEquals(excerpt($string, $phrase), '... '.$this->_long_string.' ...');
+    }
```

Comment:
Could use an additional test to ensure that it's limiting the results correctly within a radius (and preferably at word breaks?) Currently, you're only testing that it's appending an ellipsis the on the beginning and end of the string.